### ISSUE TITLE
sendToAll

### ISSUE THREAD

**Issue by rezoner** [alexhultman/uWebSockets#136]
Hiya, performance of the lib is amazing. I have one request that every lib is missing and that is crucial for games. **Method to send to every socket connected - or - to an array of sockets.**

Because we send exactly same chunks of data to 100 clients - which means the same chunk gets compressed/deflated 100 times. So what we do instead is compressing data manually before sending but then we loose the ability to use built-in browser inflate because no appropriate header is added to the message.

Basically features like perMessageDeflate are cpu killers for games at the moment.

**Comment by alexhultman** [comment]
You can disable permessage-deflate and just use regular send() or broadcast()?

I'm trying to figure out how to optimize broadcasts with permessage-deflate. If you use no context takeover for client and server you can actally pre compress one identical message and send it while still relying on the browser's built-in uncompressor (I'm almost certain).

I'm planning to have preparedMessage = prepareMessage(message) and sendPrepared(preparedMessage); or something like this to allow better perf when broadcasting or semi broadcasting.

**Comment by mafrost** [comment]
Sounds good. I do like, however, the idea of sending to an array of sockets in a more performant way than looping through them in Node and call send on each one of interest. So not really a broadcast but rather a semi broadcast as you called it (maybe a new parameter to broadcast function?). 
I guess this is more important when using the lib via node.

**Comment by alexhultman** [comment]
Will track this under #141
